test/TestUtilities/Test.Utility/PackageManagement/TestSolutionManager.cs

```diff
@@ -44,7 +45,8 @@ public TestSolutionManager(string solutionDirectory)
 
         public MSBuildNuGetProject AddNewMSBuildProject(string projectName = null, NuGetFramework projectTargetFramework = null, string packagesConfigName = null)
         {
-            if (GetNuGetProject(projectName) != null)
+            var existingProject = ThreadHelper.JoinableTaskFactory.Run(async () => await GetNuGetProjectAsync(projectName));
```

Comment:
That would simply queue a new task to run on threadpool where as here I want to synchronously execute this task.